### ISSUE TITLE
Suggestion: ship production ready build to npm

### ISSUE THREAD

**Issue by jmike** [trekhleb/javascript-algorithms#196]
Hey,

I noticed that the current version deployed on npm (i.e. https://www.npmjs.com/package/javascript-algorithms-and-data-structures) is more or less a mirror of the source code here. `javascript-algorithms` is really awesome and someone might want to use it as an external library on production (e.g. a web app or node app). Unfortunately, the use of ES modules, complicates things a bit.

May I suggest:
1. Transpiling code to ES2015, using CommonJS instead of ES modules;
2. Removing redundant source code (e.g. comments, README, etc) to reduce package size.

I'd be more that willing to provide a PR using rollup + babel.

**Comment by trekhleb** [comment]
Hi @jmike 
Thank you for suggestion.

Please go ahead and create a PR for that if you want to.

One important thing that should probably be mentioned here is that this repo is not meant to be used on production. It is for learning and researching purposes only. The only reason for having this repo on npm is probably to make it possible to re-use its function by other non-production research projects.

**Comment by jasonwilliams** [comment]
I agree with @trekhleb i'm not sure its a good idea to advertise these functions for production use, it might set a precedent of users then wanting to optimise certain functions, which then defeats the learning aspect of it. 
Maybe its worth saying in the README these shouldn't be used in production instead?

**Comment by trekhleb** [comment]
@jasonwilliams this notification it is already in README [here](https://github.com/trekhleb/javascript-algorithms#javascript-algorithms-and-data-structures).

**Comment by jasonwilliams** [comment]
ahh i should have RTFM!

**Comment by trekhleb** [comment]
No problem :D

**Comment by JXWJS** [comment]
J

**Comment by mik-laj** [comment]
@jmike Have you made any progress in this issues? Recently, I worked on a college project and the lack of a proper package in npm made my work very difficult.

**Comment by moodmosaic** [comment]
What would be the easiest to reference this project from another project? Cloning the repo and running `npm install` didn't help much:

```js
require(
  "../node_modules/javascript-algorithms-and-data-structures/src/algorithms/graph/bellman-ford/bellmanFord.js")
```

```
export default function bellmanFord(graph, startVertex) {
^^^^^^

SyntaxError: Unexpected token 'export'
```
